### ISSUE TITLE
Rename config section [workflow_engine] to [workflowengine]

### ISSUE THREAD

**Issue by nmaludy** [StackStorm/st2#4316]
##### SUMMARY

To be consistent with other config sections, the `[workflow_engine]` section should be renamed to `[workflowengine]`

##### ISSUE TYPE
 - Feature Idea

##### STACKSTORM VERSION
```
$ st2 --version
st2 2.8.1, on Python 2.7.5
```

##### EXPECTED RESULTS
The `[workflowengine]` section name is used for Orquesta config options.

**Comment by Kami** [comment]
Agreed, I'm not a fan of those inconsistencies myself (we have a similar story for timersengine, IIRC).

I would prefer to use underscore everywhere, but this would mean more work and more changes so no underscores is probably a way to go forward...

**Comment by armab** [comment]
+1

Even considering that @Kami and @m4dcoder are for using `_` underscore in naming and @lakshmi-kannan is against that, I now have decent argument for _not_ using underscore.

The example is simple: iterating on `workflowengine timersengine` should work fine with `st2.conf` sections. Adding `st2` prefix to that should produce name for service/component name.

Eg. when we name service name `st2workflowengine`, we should use `st2.conf` section naming in a similar way `workflowengine`, not `workflow_engine`.

**Comment by armab** [comment]
@nmaludy Yep, same idea with following st2 service name :+1: 
Seems we hit similar issues somewhere.
